lib/control/src/pid_controller.rs

```diff
@@ -0,0 +1,71 @@
+pub trait PidController {
+    fn new(config: PidGain) -> Self;
```

Comment:
i wouldn't put constructors in a trait, cos if you implement it again you will probably want to pass in other values for the constructor.

e.g. if you have a `HardcodedPidController` for testing something you might want to pass in a list of preset values for it to emit

also, the only place you would use a constructor that's behind a trait is if you have some nasty generic thing that is prob a sign to rethink what you're doing anyway